lib/http/HttpClient_Apple.mm

```diff
@@ -50,9 +52,11 @@ void SendAsync(IHttpResponseCallback* callback)
         {
             m_callback = callback;
             NSString* url = [[NSString alloc] initWithUTF8String:m_url.c_str()];
```

Comment:
If session is static and url changes overtime during session, this change would events sent to wrong endpoint

Reply:
As per the [documentation from apple](https://developer.apple.com/documentation/foundation/urlsession) 

> Your app creates one or more URLSession instances, each of which coordinates a group of related data-transfer tasks. For example, if you’re creating a web browser, your app might create one session per tab or window, or one session for interactive use and another for background downloads. Within each session, your app adds a series of tasks, each of which represents a request for a specific URL (following HTTP redirects, if necessary).

In the current implementation if the URL changes overtime, then a new URL Session Task would be created and even though if we have the same URL session, the events would still be routed to the correct endpoint because URL session can handle multiple URL Session Tasks

Reply:
@samirnen can you confirm you have tested this scenario? Getting URL updates wrong will have severe implications for Privacy and Compliance of the data being exhausted.

Reply:
> Is there a way that I can test this out

I don't think this use-case is valid as of now  - as we can't change the endpoint URL once SDK is bootstrapped.

Reply:
Okay. In that case having a single URL session should not create any issues right? Even if the URL can be changed, it should not cause issues (according to Apple documentation) but that flow cannot be tested right now.

Reply:
yes that's correct.